### PR TITLE
Remove CBOR instances for PLC

### PR DESCRIPTION
**Pull request by michaelpj**
It's risky having two serialization instances for a type, there's a risk
that someone will use the wrong one and then have difficulty
deserializing. Moreover, we don't *use* the CBOR instance anywhere any
more, we're fairly committed to `Flat`. So we might as well just get rid
of it.

We could add a CBOR instance that mirrors the instance for `Script`,
i.e. emits a CBOR bytestring containing the `Flat`-encoded program.
However, I think it's better to hang that on the `Script` type: the
"true" serialization of a PLC program is the `Flat` version, and the
`Flat`-in-CBOR thing is really just for playing nice with the ledger,
and so it fits to hang it on `Script`.

Bonus changes:
- The `nofib` exe had some commands for dealing with CBOR stuff, I
blindly changed those to be `Flat`
- Deleted the CBOR-vs-`Flat` comparison benchmark, since there is no
CBOR implementation to compare against!
- Deleted the handling for CBOR in the `plc` executables.

Pre-submit checklist:
- Branch
    - [x] Commit sequence broadly makes sense
    - [] Key commits have useful messages
    - [ ] Relevant tickets are mentioned in commit messages
    - [x] Formatting, materialized Nix files, PNG optimization, etc. are updated
- PR
    - [x] Self-reviewed the diff
    - [x] Useful pull request description
    - [x] Reviewer requested

Pre-merge checklist:
- [ ] Someone approved it
- [ ] Commits have useful messages
- [ ] Review clarifications made it into the code
- [ ] History is moderately tidy; or going to squash-merge

### CODE REVIEW

**Review by thealmarty — APPROVED** [review]
I found a few more CBOR things. I put them in a PR to this PR. I hope that doesn't break the CI etc! Otherwise LGTM. Glad you are doing this now, it'll make my upcoming tasks easier! Thanks!

File: plutus-benchmark/nofib/exe/Main.hs [hunk 1]
```diff
@@ -49,8 +49,8 @@ data Options
     = RunPLC           ProgAndArgs
     | RunHaskell       ProgAndArgs
     | DumpPLC          ProgAndArgs
-    | DumpCBORnamed    ProgAndArgs
-    | DumpCBORdeBruijn ProgAndArgs
+    | DumpFlatnamed    ProgAndArgs
```
Comment (kwxm) [inline plutus-benchmark/nofib/exe/Main.hs:52]:
```suggestion
    | DumpFlatNamed    ProgAndArgs
```

File: plutus-benchmark/nofib/exe/Main.hs [hunk 2]
```diff
@@ -49,8 +49,8 @@ data Options
...
+    | DumpFlatdeBruijn ProgAndArgs
```
Comment (kwxm) [inline plutus-benchmark/nofib/exe/Main.hs:53]:
```suggestion
    | DumpFlatDeBruijn ProgAndArgs
```

**Review by kwxm — APPROVED** [review]
This all looks good.  Since we're losing the serialisation benchmarks I ran them and recorded the results to remind us how we got to where we are: see #3529.

**Comment by michaelpj** [comment]
> I found a few more CBOR things. I put them in a PR to this PR. 

Thanks!

**Pull request merged**